DataSetExplorer/DataSetSerializer/CompleteDataSetExporter.cs

```diff
@@ -10,28 +10,38 @@ namespace DataSetExplorer.DataSetSerializer
 {
     class CompleteDataSetExporter
     {
-        private readonly string _templatePath = "../../../DataSetSerializer/Template/Complete_Dataset_Template.xlsx";
+        private readonly string _dataSetWithAnnotationsTemplatePath = "../../../DataSetSerializer/Template/Complete_Dataset_With_Annotations_Template.xlsx";
+        private readonly string _dataSetWithHeuristicsTemplatePath = "../../../DataSetSerializer/Template/Complete_Dataset_With_Heuristics_Template.xlsx";
+        private readonly string _dataSetWithMetricsTemplatePath = "../../../DataSetSerializer/Template/Complete_Dataset_With_Metrics_Template.xlsx";
         private readonly string _exportPath;
         private ExcelPackage _excelFile;
         private ExcelWorksheet _sheet;
+        private readonly ColumnHeuristicsModel _requiredSmells = new ColumnHeuristicsModel();
 
         public CompleteDataSetExporter(string exportPath)
         {
             _exportPath = exportPath;
         }
 
-        public void Export(List<DataSetInstance> dataSetInstances, string fileName)
+        public void Export(List<DataSetInstance> dataSetInstances, string smell, string fileName)
+        {
+            ExportDataSetWithAnnotations(dataSetInstances, fileName);
```

Comment:
Maybe each of these segments deserves a separate class. I would say "no" at this moment, but if the complexity of this class continues to grow (and surpasses roughly 250 LOC) I would consider splitting the class, especially if there is no or very little shared logic.

On a related matter, the `Export`, `ExportDataSetWithAnnotations`, `ExportDataSetWithMetrics`, and `ExportDataSetWithHeuristics `roughly have the same API. If we go through with the extractions, it might make sense to introduce an interface and have the `CompleteDataSetExporter` implement it and maintain a list of `IDatasetExporter`s.

Reply:
I was thinking the same. I will keep this in mind.